sunpy/image/resample.py

```diff
@@ -54,6 +56,8 @@ def resample(orig, dimensions, method='linear', center=False, minusone=False):
     if len(dimensions) != orig.ndim:
         raise UnequalNumDimensions("Number of dimensions must remain the same "
                                    "when calling resample.")
+    if method == 'spline' and (np.any(np.isnan(orig)) or np.any(np.isinf(orig))):
```

Comment:
```suggestion
    if method == 'spline' and not np.isfinite(orig).all():
```
`isfinite` returns False if the value is NaN or Inf